Redesign home-tab GSoC M1.3,M1.4

SubhashThenua opened the pull request:
## Overview
<!--
READ ME FIRST:
Please answer *both* questions below and check off every point from the Essential Checklist!
If there is no corresponding issue number, fill in N/A where it says [fill_in_number_here] below in 1.
-->

1. This PR fixes or fixes part of #18384 .
2. This PR does the following: **1)** This PR redesign the Home-tab of learner dashboard as per the mocks(https://www.figma.com/file/YWe7SqfUVjxlJLKTUn0UZa/Project-2?type=design&node-id=3248%3A27299&mode=design&t=k0A4DAIPJFBohCBb-1).
3. (For bug-fixing PRs only) The original bug occurred because: [Explain what
   the cause of the bug was, and which PR introduced it]

## Essential Checklist

- [ ] The **PR title** starts with "Fix #bugnum: " or "Fix part of #bugnum: ...",
followed by a short, clear summary of the changes.
- [x] I have followed the [instructions for making a code change](https://github.com/oppia/oppia/wiki/Make-a-pull-request).
- [ ] The linter/Karma presubmit checks have passed on my local machine.
- [x] "Allow edits from maintainers" is checked. (See [here](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork) for instructions on how to enable it.)
- [ ] I have assigned the correct reviewers to this PR (or will leave a
comment with the phrase "@{{reviewer_username}} PTAL" if I don't have
permissions to assign reviewers directly).

## Proof that changes are correct


[3rd.webm](https://github.com/oppia/oppia/assets/100484672/630f8629-a3b6-4418-95ee-9638bcd82e12)

E2E Test for Milestone_1

[e2e_for_GSoC_Mile_1.webm](https://github.com/oppia/oppia/assets/100484672/2b747747-9ff8-40ca-b027-399c16e01719)



## PR Pointers

- Never force push! If you do, your PR will be closed.
- Make sure your PR follows the [style guide](https://github.com/oppia/oppia/wiki/Coding-style-guide), otherwise this will lead to review delays.
- Some e2e tests are flaky, and can fail for reasons unrelated to your PR. We are working on fixing this, but in the meantime, if you need to restart the tests, please check the ["If your build fails" wiki page](https://github.com/oppia/oppia/wiki/If-CI-checks-fail-on-your-PR).
- See the [Code Owner's wiki page](https://github.com/oppia/oppia/wiki/Oppia's-code-owners-and-checks-to-be-carried-out-by-developers) for what code owners will expect.

Commit:
new-learner-dashboard-component-created

Commit:
new-learner-dashboard-minor changes

Commit:
minor changes

Commit:
redesign the tiles

Commit:
fixing the card UI

Commit:
created circular-bar

Commit:
added progress circle

Commit:
minor changes

Commit:
Merge remote-tracking branch 'upstream/develop' into Redesign-home-tab-GSoC-M1.2

Commit:
concept to seprate the chapter in progress and recommended and mark the tile as goal

Commit:
exploration summary tile for learner dashboard

Commit:
redesigned collection tile

Commit:
added community lessons playlist in home-tab

Commit:
Merge remote-tracking branch 'upstream/develop' into Redesign-home-tab-GSoC-M1.2

Commit:
resolve conflicts

Commit:
fixes the UI of exploraton card

Commit:
created carousel files

Commit:
fix the carousel error

Commit:
Merge remote-tracking branch 'upstream/develop' into Redesign-home-tab-GSoC-M1.2

Commit:
changes to carousel

Commit:
carousel fixes

Commit:
Merge remote-tracking branch 'upstream/develop' into Redesign-home-tab-GSoC-M1.2

Commit:
revert the changes of #18406 (beam job of kshitij)

Commit:
added scroll bar with buttons

Commit:
added math classroom button

Commit:
fix carousel error

Commit:
Merge remote-tracking branch 'upstream/develop' into Redesign-home-tab-GSoC-M1.2

Commit:
fix carousel issues

Commit:
make responsive

Commit:
add search bar

Commit:
Merge remote-tracking branch 'upstream/develop' into Redesign-home-tab-GSoC-M1.2

Commit:
search-bar-responsive

Commit:
search-bar-for-mobile-view-done

Commit:
make playlist responsive

Commit:
Merge remote-tracking branch 'upstream/develop' into Redesign-home-tab-GSoC-M1.3

Commit:
added story ids to learner group summary

Commit:
added learner group title to story summary data

Commit:
fix search bar css

Commit:
Merge remote-tracking branch 'upstream/develop' into Redesign-home-tab-GSoC-M1.3

Commit:
added the data to the story tile

Commit:
added exploration checkpoints fetcher

Commit:
fix story card view

Commit:
fix exploration card view

Commit:
minor changes

Commit:
Merge remote-tracking branch 'upstream/develop' into Redesign-home-tab-GSoC-M1.3

SubhashThenua committed:
Merge branch 'develop' into Redesign-home-tab-GSoC-M1.3

Commit:
added progress calculation

Commit:
added exploration to continue-where-you-left

Commit:
Merge branch 'Redesign-home-tab-GSoC-M1.3' of https://github.com/SubhashThenua/oppia into Redesign-home-tab-GSoC-M1.3

Commit:
Merge remote-tracking branch 'upstream/develop' into Redesign-home-tab-GSoC-M1.3

Commit:
Merge remote-tracking branch 'upstream/develop' into Redesign-home-tab-GSoC-M1.3

Commit:
making responsive

Commit:
Merge remote-tracking branch 'upstream/develop' into Redesign-home-tab-GSoC-M1.3

Commit:
making responsive

Commit:
Merge remote-tracking branch 'upstream/develop' into Redesign-home-tab-GSoC-M1.3

Commit:
responsive search bar

Commit:
Merge remote-tracking branch 'upstream/develop' into Redesign-home-tab-GSoC-M1.3

Commit:
added Progress for exp

Commit:
redesign-learner-groups-tab

Commit:
remove unwanted code

SubhashThenua committed:
Update constants.ts

Commit:
removed unwanted code

Commit:
Merge branch 'Redesign-home-tab-GSoC-M1.3' of https://github.com/SubhashThenua/oppia into Redesign-home-tab-GSoC-M1.3

SubhashThenua committed:
Update valid.py

Commit:
redesign tablet view buttons menu

Commit:
redesign learner dashboard menu

Commit:
fix collection card

Commit:
Merge remote-tracking branch 'upstream/develop' into Redesign-home-tab-GSoC-M1.3

Commit:
added progress calculation for collections

Commit:
minor changes

Commit:
Merge remote-tracking branch 'upstream/develop' into Redesign-home-tab-GSoC-M1.3

SubhashThenua committed:
Update constants.ts

Commit:
fix carousel

Commit:
Merge branch 'Redesign-home-tab-GSoC-M1.3' of https://github.com/SubhashThenua/oppia into Redesign-home-tab-GSoC-M1.3

Commit:
Merge remote-tracking branch 'upstream/develop' into Redesign-home-tab-GSoC-M1.3

heyimShivam commented:
Hi @SubhashThenua The ui does not seem close enough to the mocks, the padding is large, and the font size is not the same, please check UI once again for all components (tags, underline, etc). the website should be the mirror image of the mocks at least 98% the same.

![Screenshot 2023-07-14 142207](https://github.com/oppia/oppia/assets/77006897/9df48301-419d-456e-bbeb-81ed22eddb9d)

heyimShivam commented on core/templates/pages/learner-dashboard-page/learner-groups-tab.component.ts:
```diff
@@ -64,6 +64,7 @@ export class LearnerGroupsTabComponent {
         (learnerDashboardLearnerGroups) => {
           this.learnerGroupsJoined = (
             learnerDashboardLearnerGroups.learnerGroupsJoined);
+          console.error(this.learnerGroupsJoined, 'group joined bhole');
           this.invitedToLearnerGroups = (
```
Please evaluate your work yourself before requesting a review. here and elsewhere!!.

SubhashThenua replied:
Removed
Thanks!

Commit:
added goal and learner group title tags on story

Commit:
added unit test for carousel

kevintab95 commented on core/domain/exp_domain.py:
```diff
@@ -6723,3 +6723,12 @@ def to_dict(self) -> ExplorationVersionHistoryDict:
+
+    exploration_id: str
+    visited_checkpoints_count: int
+    total_checkpoints_count: int
```
Please add a newline.

kevintab95 replied:
Ditto elsewhere (where this icon is shown in the PR.)

SubhashThenua replied:
Done

kevintab95 commented on core/templates/components/summary-tile/collection-summary-tile.component.ts:
```diff
@@ -127,6 +184,13 @@ export class CollectionSummaryTileComponent implements OnInit, OnDestroy {
+    if (this.buttonIsHovered) {
+      return 'box-shadow: 0 5px 5px -5px #333;';
+    }
+    // Return 'height: 144px; width: 192px;';
```
Please avoid commented code.

SubhashThenua replied:
I have already removed. PTAL

Commit:
fixing the css and test

Commit:
Merge remote-tracking branch 'upstream/develop' into Redesign-home-tab-GSoC-M1.3

Commit:
fix all the carosel front end tests

Commit:
Merge remote-tracking branch 'upstream/develop' into Redesign-home-tab-GSoC-M1.3

Commit:
nit changes

SubhashThenua committed:
Update carousel-bar.component.html

Commit:
added learner dashboard service

Commit:
changes in leaner dashbaod data status service

Commit:
fix the delay in passing the data to home tab

Commit:
removed unused service, created by me

Commit:
added I18N

Commit:
added I18N and fix responsive

Commit:
Merge branch 'Redesign-home-tab-GSoC-M1.3' of https://github.com/SubhashThenua/oppia into Redesign-home-tab-GSoC-M1.3

SubhashThenua committed:
Merge branch 'develop' into Redesign-home-tab-GSoC-M1.3

Commit:
added new hometab

Commit:
old home tab back

Commit:
added feature flag

Commit:
nit changes

Commit:
Merge branch 'Redesign-home-tab-GSoC-M1.3' of https://github.com/SubhashThenua/oppia into Redesign-home-tab-GSoC-M1.3

SubhashThenua committed:
Merge branch 'develop' into Redesign-home-tab-GSoC-M1.3

Commit:
fixed dashboard menu

Commit:
nit changes

Commit:
Merge branch 'Redesign-home-tab-GSoC-M1.3' of https://github.com/SubhashThenua/oppia into Redesign-home-tab-GSoC-M1.3

SubhashThenua committed:
Update learner-groups-tab.component.ts

SubhashThenua committed:
Update collection-player-page.component.ts

SubhashThenua commented:
> Hi @SubhashThenua The ui does not seem close enough to the mocks, the padding is large, and the font size is not the same, please check UI once again for all components (tags, underline, etc). the website should be the mirror image of the mocks at least 98% the same.
> 
> ![Screenshot 2023-07-14 142207](https://user-images.githubusercontent.com/77006897/253596709-9df48301-419d-456e-bbeb-81ed22eddb9d.jpg)

I added new videos
PTAL
Thanks!

Commit:
removed unwanted code

Commit:
Merge branch 'Redesign-home-tab-GSoC-M1.3' of https://github.com/SubhashThenua/oppia into Redesign-home-tab-GSoC-M1.3

aasiffaizal commented:
Could you reassign me once the tests are fixed? @SubhashThenua

Commit:
nit changes

Lawful2002 commented:
Please reassign when frontend tests are fixed.

heyimShivam commented:
@SubhashThenua Please re-upload the Proof video again as some of the components are not showing images and make sure everything looks like Figma design no extra padding or margin.

heyimShivam commented:
search bar is still not up to the mark.

heyimShivam commented on core/templates/services/contextual/window-dimensions.service.ts:
```diff
@@ -49,6 +49,11 @@ export class WindowDimensionsService {
     let NORMAL_NAVBAR_CUTOFF_WIDTH_PX = 768;
     return this.getWidth() <= NORMAL_NAVBAR_CUTOFF_WIDTH_PX;
   }
+
+  isTabletView(): boolean {
+    let TABLET_LANDSCAPE_CUTOFF_WIDTH_PX = 1024;
+    return this.getWidth() <= TABLET_LANDSCAPE_CUTOFF_WIDTH_PX;
+  }
 }
```
Why is this required?

SubhashThenua replied:
We did not have any function to know whether the tablet landscape view is active or not. As per the mocks design I need it.

heyimShivam commented on core/templates/pages/library-page/search-bar/search-bar.component.ts:
```diff
@@ -89,7 +90,7 @@ export class SearchBarComponent implements OnInit, OnDestroy {
   }
 
   isMobileViewActive(): boolean {
-    return this.windowDimensionsService.getWidth() <= 766;
+    return this.windowDimensionsService.getWidth() <= 639;
```
?

SubhashThenua replied:
In the Figma design, we are changing the search bar from desktop view to mobile view at 640px.

SubhashThenua replied:
Hi, @heyimShivam I find a new way that does not require to modify it.
Thanks!

heyimShivam commented on core/templates/pages/library-page/search-bar/search-bar.component.css:
```diff
@@ -277,19 +298,52 @@ oppia-search-bar .oppia-input-group {
   }
 }
 
+/* @media (max-width: 1600px) {
+  oppia-search-bar .oppia-input-group {
+    width: 52.5em;
+  }
+} */
+
```
please only assign reviewers, when you review your pr first by yourself. (Please remove the commented code)

SubhashThenua replied:
Done PTAL

heyimShivam commented:
https://github.com/oppia/oppia/pull/18594#discussion_r1263851222 Don't ignore this

Commit:
added frontend test for new-home-tab

Commit:
removed commented code

Commit:
Merge remote-tracking branch 'upstream/develop' into Redesign-home-tab-GSoC-M1.3

Commit:
added test

Commit:
fix search bar

Commit:
fix test

Commit:
added unit test for new leanerdashboard component

SubhashThenua committed:
Merge branch 'develop' into Redesign-home-tab-GSoC-M1.3

Commit:
fix frontend test

Commit:
fix test

Commit:
fix coverage

Commit:
fix typescripts

Commit:
fix linter of new-leaner-dashboard

Commit:
Merge remote-tracking branch 'upstream/develop' into Redesign-home-tab-GSoC-M1.3

Commit:
fix unit test and typescripts test

Commit:
fix unit,lint,backend tests

Commit:
Merge remote-tracking branch 'upstream/develop' into Redesign-home-tab-GSoC-M1.3

Commit:
fix mypy

Commit:
fix search bar

Commit:
fix linter

SubhashThenua committed:
Merge branch 'develop' into Redesign-home-tab-GSoC-M1.3

SubhashThenua commented:
Hi, @heyimShivam I have updated PR and added the 3rd video PTAL 
Thanks!

heyimShivam commented:
> Hi, @heyimShivam I have updated PR and added the 3rd video PTAL Thanks!

Please remove the old videos if they are no longer needed.

heyimShivam commented on assets/i18n/fi.json:
```diff
@@ -610,7 +610,6 @@
 	"I18N_LEARNER_GROUP_SYLLABUS_ITEM_NOT_STARTED_YET": "Ei vielä aloitettu",
 	"I18N_LEARNER_GROUP_SYLLABUS_LESSONS": "opoitunnit",
 	"I18N_LEARNER_GROUP_SYLLABUS_TAB": "Opetussuunnitelma",
-	"I18N_LEARNER_GROUP_VIEW_DETAILS": "Näytä tiedot",
 	"I18N_LEARNER_GROUP_VIEW_PREFERENCES": "RYHMÄN ASETUKSET",
```
Why didn't you include more variables in this i18n file +  in other i18n files?  eg: I18N_LEARNER_GROUP_MORE_INFO, I18N_LEARNER_GROUP_DECLINE_INVITAITON etc.

SubhashThenua replied:
Hi, @heyimShivam  Thanks for review.
1) I18N_LEARNER_GROUP_VIEW_DETAILS is now changed to I18N_LEARNER_GROUP_MORE_INFO so I have removed the view-detail and instead of i used more info as per the mocks.
2)FYI, When we include the new I18N string , we only include in en.json and qqq.json and rest of the files were modified by others.

SubhashThenua commented:
> > Hi, @heyimShivam I have updated PR and added the 3rd video PTAL Thanks!
> 
> Please remove the old videos if they are no longer needed.

I think it is better to have them also.

seanlip commented:
Just a note, you only need videos corresponding to the latest version of
the code, otherwise you'll confuse the reviewers.

If you need more than one video to show the functionality for the latest
version of the code, that's fine. But don't include videos for previous
versions of the code.

On Tue, Jul 25, 2023, 1:09 PM SUBHASH THENUA ***@***.***>
wrote:

> Hi, @heyimShivam <https://github.com/heyimShivam> I have updated PR and
> added the 3rd video PTAL Thanks!
>
> Please remove the old videos if they are no longer needed.
>
> I think it is better to have them also.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/oppia/oppia/pull/18594#issuecomment-1649133859>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACQV5SRDXNFSWNRASWYHJ4DXR5IKLANCNFSM6AAAAAA2A3V6HQ>
> .
> You are receiving this because your review was requested.Message ID:
> ***@***.***>
>

Lawful2002 commented on core/templates/components/summary-tile/exploration-summary-tile.component.spec.ts:
```diff
@@ -533,4 +536,39 @@ describe('Exploration Summary Tile Component', () => {
+  it('should set the getStrokeDashOffSet', fakeAsync(() => {
+    const explorationProgress = 75;
+    const expectedOffsetValue =
+    (
+      component.circumference -
+      (explorationProgress / 100) * component.circumference);
```
```suggestion
    (
      component.circumference -
      (explorationProgress / 100) * component.circumference
    );
```

Lawful2002 replied:
Improves readability.

SubhashThenua committed:
Merge branch 'develop' into Redesign-home-tab-GSoC-M1.3

Lawful2002 commented on core/templates/pages/learner-dashboard-page/new-learner-dashboard.component.spec.ts:
```diff
@@ -0,0 +1,883 @@
+        return Promise.resolve('sample-csrf-token');
+      });
+      // Generate completed explorations and exploration playlist.
+      for (let i = 0; i < 10; i++) {
```
What is the significance of `10` here? Would it be better to add it as a variable?

SubhashThenua replied:
Hi, @Lawful2002 Thanks for reviewing. I have created **new-learner-dashboard.ts/spec.ts**  copy of **learner-dashboard.ts/spec.ts** to maintain the existing learner-dashboard and redesigned learner-dashboard. So I have not written this test. It is already written Please see here (https://github.com/SubhashThenua/oppia/blob/develop/core/templates/pages/learner-dashboard-page/learner-dashboard-page.component.spec.ts#L350)
Thanks!

Lawful2002 commented on core/templates/pages/learner-dashboard-page/new-learner-dashboard.component.spec.ts:
```diff
@@ -0,0 +1,883 @@
+
+    it('should get formatted date string from the timestamp in milliseconds',
+      () => {
+        // This corresponds to Fri, 2 Apr 2021 09:45:00 GMT.
```
Would it be better to generate a random date?

SubhashThenua replied:
Same here please see (https://github.com/SubhashThenua/oppia/blob/develop/core/templates/pages/learner-dashboard-page/learner-dashboard-page.component.spec.ts#L615)

Lawful2002 replied:
@kevintab95 Should we use a random date (created on runtime) for this test, or should we go with this? I suppose this approach is fine as well.

Lawful2002 reviewed (changes requested):
Thanks @SubhashThenua, a few comments

SubhashThenua committed:
Update core/templates/components/summary-tile/exploration-summary-tile.component.spec.ts

Co-authored-by: Harshvardhan Singh <73544247+Lawful2002@users.noreply.github.com>

Commit:
Merge remote-tracking branch 'upstream/develop' into Redesign-home-tab-GSoC-M1.3

heyimShivam commented on core/templates/pages/library-page/search-bar/search-bar.component.ts:
```diff
@@ -70,6 +70,7 @@ export class SearchBarComponent implements OnInit, OnDestroy {
   translationData: Record<string, number> = {};
   activeMenuName: string = '';
   @Input() enableDropup: boolean = false;
+  @Input() learnerDashboardSearch: boolean = false;
```
have you used this new variable?

SubhashThenua replied:
Removed as unused.

heyimShivam commented on core/templates/pages/learner-dashboard-page/new-learner-dashboard.component.ts:
```diff
@@ -0,0 +1,388 @@
+  allTopics!: LearnerTopicSummary[];
+  untrackedTopics!: Record<string, LearnerTopicSummary[]>;
+  subscriptionsList!: ProfileSummary[];
+
```
```suggestion
```

heyimShivam commented on core/templates/pages/learner-dashboard-page/new-learner-dashboard.component.ts:
```diff
@@ -0,0 +1,388 @@
+  storyIdToLearnerGroupsTitleMap: Map<string, string[]> = new Map();
+  activeSection!: string;
+  activeSubsection!: string;
+
```
```suggestion
```

heyimShivam commented on core/templates/pages/learner-dashboard-page/new-learner-dashboard.component.ts:
```diff
@@ -0,0 +1,388 @@
+  explorationId!: string;
+  communityLibraryUrl = (
+    '/' + AppConstants.PAGES_REGISTERED_WITH_FRONTEND.LIBRARY_INDEX.ROUTE);
+
```
```suggestion
```

please correct it everywhere, no new line is required while creating variables.

heyimShivam commented on core/templates/pages/learner-dashboard-page/new-learner-dashboard.component.ts:
```diff
@@ -0,0 +1,388 @@
+          (learnerDashboardLearnerGroups) => {
+            let learnerGroupsJoined = (
+              learnerDashboardLearnerGroups.learnerGroupsJoined);
+            for (var learnerGroup of learnerGroupsJoined) {
```
```suggestion
            for (let learnerGroup of learnerGroupsJoined) {
```
use let on the place of var. here and elsewhere

heyimShivam commented on core/templates/pages/learner-dashboard-page/new-learner-dashboard.component.ts:
```diff
@@ -0,0 +1,388 @@
+              let storyIds = learnerGroup.storyIds;
+              for (var storyId of storyIds) {
+                if (this.storyIdToLearnerGroupsTitleMap.has(storyId)) {
+                  this.storyIdToLearnerGroupsTitleMap.
+                    get(storyId)?.push(title);
+                } else {
```
please remove `?` here and elsewhere.

SubhashThenua replied:
Using ? to make the typescripts strict check. I have discussed this with kevin.

heyimShivam replied:
please remove var and try to use let.

heyimShivam commented on core/templates/pages/learner-dashboard-page/new-learner-dashboard.component.ts:
```diff
@@ -0,0 +1,388 @@
+      dashboardCollectionsDataPromise,
+      dashboardExplorationsDataPromise,
+    ]).then(() => {
+      setTimeout(() => {
+        this.loaderService.hideLoadingScreen();
+        this.activeSection = (
+          LearnerDashboardPageConstants
+            .LEARNER_DASHBOARD_SECTION_I18N_IDS.HOME);
+      }, 0);
```
please remove settimeout or add a comment about why you have used it here.

SubhashThenua replied:
Already used please see (https://github.com/oppia/oppia/blob/develop/core/templates/pages/learner-dashboard-page/learner-dashboard-page.component.ts#L239). Also we are using setTimeout to wait for all the promise get resolve.

heyimShivam replied:
does Promise.all is not doing the same thing?

SubhashThenua replied:
Hi, @heyimShivam Thanks for review, As I have already mentioned, it used at many places in the codebase and I agree with  (does Promise.all is not doing the same thing?) but I would like to mention one point why I need setTimeout.
1) Learner dashboard page.html contains home-tab.component and home-tab will only render when 
activeSection = LearnerDashboardPageConstants .LEARNER_DASHBOARD_SECTION_I18N_IDS.HOME);

- Sometimes promise get resolve before the home-tab render (That's no issue)

- But sometimes home-tab get render before the promise resolve and this cause the issues (That is data is not passed to component).

heyimShivam commented on core/templates/pages/learner-dashboard-page/new-learner-dashboard.component.ts:
```diff
@@ -0,0 +1,388 @@
+    this.pageTitleService.setDocumentTitle(translatedTitle);
+  }
+
+
```
```suggestion

```
Please check your  PR carefully before asking for reviews, and try not to repeat the same things again.

heyimShivam commented on core/templates/pages/learner-dashboard-page/new-learner-dashboard.component.spec.ts:
```diff
@@ -0,0 +1,883 @@
+ * @fileoverview Unit tests for learner dashboard parge.
+ */
+
+
```
```suggestion

```

heyimShivam commented on core/templates/pages/learner-dashboard-page/new-learner-dashboard.component.spec.ts:
```diff
@@ -0,0 +1,883 @@
+import { Collection, CollectionBackendDict } from 'domain/collection/collection.model';
+import { LearnerExplorationSummary } from 'domain/summary/learner-exploration-summary.model';
+
+
```
```suggestion
```

heyimShivam commented on core/templates/pages/learner-dashboard-page/new-learner-dashboard.component.spec.ts:
```diff
@@ -0,0 +1,883 @@
+import { HttpClientTestingModule } from '@angular/common/http/testing';
+import { Component, EventEmitter, NO_ERRORS_SCHEMA, Pipe } from '@angular/core';
+import { TranslateService } from '@ngx-translate/core';
+
```
```suggestion
```

heyimShivam commented on core/templates/pages/learner-dashboard-page/new-learner-dashboard.component.spec.ts:
```diff
@@ -0,0 +1,883 @@
+    }
+  };
+
+
```
```suggestion

```

heyimShivam commented on core/templates/pages/learner-dashboard-page/new-learner-dashboard.component.spec.ts:
```diff
@@ -0,0 +1,883 @@
+    collection_playlist: []
+  };
+
+
```
```suggestion

```

heyimShivam commented on core/templates/pages/learner-dashboard-page/new-learner-dashboard.component.spec.ts:
```diff
@@ -0,0 +1,883 @@
+      }
+
+      // Generate incomplete explorations and incomplete exploration playlist.
+      for (let i = 0; i < 12; i++) {
```
same as above.

SubhashThenua replied:
I have answered above.

heyimShivam commented on core/templates/pages/learner-dashboard-page/new-learner-dashboard.component.spec.ts:
```diff
@@ -0,0 +1,883 @@
+
+      expect(component.windowIsNarrow).toBeTrue();
+      expect(component.tabletView).toBeTrue();
+
+
```
```suggestion

```

heyimShivam commented on core/templates/pages/learner-dashboard-page/new-home-tab.component.ts:
```diff
@@ -0,0 +1,263 @@
+    this.storyExplorationProgressList.push(
+      ...this.storyInProgress, ...this.incompleteExplorationsList);
+
+    for (var exp of this.completedExplorationsList) {
```
use  let

SubhashThenua replied:
Done

heyimShivam replied:
still var

heyimShivam commented on core/templates/pages/learner-dashboard-page/new-home-tab.component.ts:
```diff
@@ -0,0 +1,263 @@
+        currentGoalsIds.push(goal.id);
+      }
+    }
+    for (var topicSummaryTile of this.continueWhereYouLeftOffList) {
```
use let

SubhashThenua replied:
Done

heyimShivam reviewed (changes requested):
Please check your PR yourself and then I will re-review your pr as soon as you have fixed all the changes in the whole pr.

SubhashThenua committed:
Update core/templates/pages/learner-dashboard-page/new-learner-dashboard.component.ts

Co-authored-by: Shivam Chaudhary <77006897+heyimShivam@users.noreply.github.com>

SubhashThenua committed:
Update core/templates/pages/learner-dashboard-page/new-learner-dashboard.component.ts

Co-authored-by: Shivam Chaudhary <77006897+heyimShivam@users.noreply.github.com>

SubhashThenua committed:
Update core/templates/pages/learner-dashboard-page/new-learner-dashboard.component.ts

Co-authored-by: Shivam Chaudhary <77006897+heyimShivam@users.noreply.github.com>

SubhashThenua committed:
Update core/templates/pages/learner-dashboard-page/new-learner-dashboard.component.spec.ts

Co-authored-by: Shivam Chaudhary <77006897+heyimShivam@users.noreply.github.com>

SubhashThenua committed:
Update core/templates/pages/learner-dashboard-page/new-learner-dashboard.component.spec.ts

Co-authored-by: Shivam Chaudhary <77006897+heyimShivam@users.noreply.github.com>

SubhashThenua committed:
Update core/templates/pages/learner-dashboard-page/new-learner-dashboard.component.ts

Co-authored-by: Shivam Chaudhary <77006897+heyimShivam@users.noreply.github.com>

SubhashThenua committed:
Update core/templates/pages/learner-dashboard-page/new-learner-dashboard.component.ts

Co-authored-by: Shivam Chaudhary <77006897+heyimShivam@users.noreply.github.com>

SubhashThenua committed:
Update core/templates/pages/learner-dashboard-page/new-learner-dashboard.component.spec.ts

Co-authored-by: Shivam Chaudhary <77006897+heyimShivam@users.noreply.github.com>

SubhashThenua committed:
Update core/templates/pages/learner-dashboard-page/new-learner-dashboard.component.spec.ts

Co-authored-by: Shivam Chaudhary <77006897+heyimShivam@users.noreply.github.com>

SubhashThenua committed:
Update core/templates/pages/learner-dashboard-page/new-learner-dashboard.component.spec.ts

Co-authored-by: Shivam Chaudhary <77006897+heyimShivam@users.noreply.github.com>

SubhashThenua committed:
Update core/templates/pages/learner-dashboard-page/new-learner-dashboard.component.spec.ts

Co-authored-by: Shivam Chaudhary <77006897+heyimShivam@users.noreply.github.com>

SubhashThenua committed:
Merge branch 'develop' into Redesign-home-tab-GSoC-M1.3

Commit:
PR review changes

Commit:
Merge branch 'Redesign-home-tab-GSoC-M1.3' of https://github.com/SubhashThenua/oppia into Redesign-home-tab-GSoC-M1.3

kevintab95 commented on assets/constants.ts:
```diff
@@ -7092,6 +7094,6 @@ export default {
   "CONTRIBUTOR_CERTIFICATE_HEIGHT": 1313,
-  "BRANCH_NAME": "",
-  "SHORT_COMMIT_HASH": ""
+  "BRANCH_NAME": "Redesign-home-tab-GSoC-M1.3",
+  "SHORT_COMMIT_HASH": "9125836a7d"
```
Please undo.

SubhashThenua replied:
Done

kevintab95 commented on core/templates/components/summary-tile/collection-summary-tile.component.css:
```diff
@@ -0,0 +1,233 @@
+}
+
+oppia-collection-summary-tile .mobile-activity-summary-card {
+  box-shadow: slategrey 2px 4px 4px 0;
```
Prefer RGB or HEX values for colours.

SubhashThenua replied:
It is already written. please see (https://github.com/SubhashThenua/oppia/blob/develop/core/templates/components/summary-tile/collection-summary-tile.component.html#L121). But I have changed.

kevintab95 commented on core/templates/components/summary-tile/collection-summary-tile.component.css:
```diff
@@ -0,0 +1,233 @@
+
+oppia-collection-summary-tile .mobile-activity-summary-card {
+  box-shadow: slategrey 2px 4px 4px 0;
+  padding: 0 0 0 0;
```
```suggestion
  padding: 0;
```

kevintab95 commented on core/templates/components/summary-tile/collection-summary-tile.component.css:
```diff
@@ -0,0 +1,233 @@
+}
+
+oppia-collection-summary-tile .mobile-collection-card-image-container {
+  box-shadow: slategrey -1px 3px 4px 0;
```
Ditto re colours.

SubhashThenua replied:
Done

kevintab95 commented on core/templates/components/summary-tile/collection-summary-tile.component.css:
```diff
@@ -0,0 +1,233 @@
+}
+
+oppia-collection-summary-tile .continue-button {
+  background: white 0% 0% no-repeat padding-box;
```
Ditto re colours.

SubhashThenua replied:
Done

kevintab95 replied:
Not done?

kevintab95 commented on core/templates/components/summary-tile/exploration-summary-tile.component.css:
```diff
@@ -69,3 +69,169 @@ oppia-exploration-summary-tile .views-text {
+oppia-exploration-summary-tile .oppia-exploration-objective-title {
+  -webkit-box-orient: vertical;
+  color: #333;
+  display: -webkit-box;
```
What does "webkit-box" do and why is it used here? I can't find documentation for it online.

SubhashThenua replied:
Removed webkit-box

kevintab95 commented on core/templates/components/summary-tile/exploration-summary-tile.component.css:
```diff
@@ -69,3 +69,169 @@ oppia-exploration-summary-tile .views-text {
+  word-wrap: break-word;
+}
+oppia-exploration-summary-tile .continue-button {
+  background: white 0% 0% no-repeat padding-box;
```
Ditto re colours.

kevintab95 replied:
Ditto elsewhere.

SubhashThenua replied:
Done

kevintab95 commented on core/templates/components/summary-tile/exploration-summary-tile.component.css:
```diff
@@ -69,3 +69,169 @@ oppia-exploration-summary-tile .views-text {
+}
+oppia-exploration-summary-tile .oppia-learner-dashboard-exploration-title {
+  color: #00645c;
+  font-family: Capriola,Roboto,Arial,sans-serif;
```
Please add spaces after ","

SubhashThenua replied:
Done

kevintab95 commented on core/templates/pages/learner-dashboard-page/carousel-bar.component.css:
```diff
@@ -0,0 +1,67 @@
+.oppia-carousel-bar .oppia-carousel-heading {
+  color: #333;
+  font-family: Roboto,sans-serif;
```
Ditto here and elsewhere.

SubhashThenua replied:
Done

kevintab95 replied:
Not done?

kevintab95 commented on core/templates/pages/learner-dashboard-page/carousel-bar.component.css:
```diff
@@ -0,0 +1,67 @@
+  overflow: hidden;
+}
+
+.oppia-carousel-bar .oppia-carousel-content::-webkit-scrollbar {
```
I see a lot of "webkit" rules being used, why is that? Should we check if the UI looks the same on Safari and Chome?

SubhashThenua replied:
It is already used at many places in code base.

kevintab95 replied:
Just to check, have you seen the second part of my comment?

kevintab95 commented on core/templates/pages/learner-dashboard-page/carousel-bar.component.css:
```diff
@@ -0,0 +1,67 @@
+  float: left;
+  height: 24px;
+  justify-content: center;
+  outline: none !important;
```
Why "!important" (see [guide](https://github.com/oppia/oppia/wiki/Coding-style-guide#css) re usage)?

SubhashThenua replied:
Removed important

kevintab95 commented on core/templates/pages/learner-dashboard-page/new-learner-dashboard.component.html:
```diff
@@ -0,0 +1,138 @@
+    </div>
+  </div>
+</div>
+<style>
+  .oppia-profile-alignment {
+    vertical-align: middle;
+  }
+  .oppia-box-size {
+    padding: 2px 10px;
+  }
+</style>
```
Should this be moved to a CSS file?

SubhashThenua replied:
Done

kevintab95 commented on core/templates/components/summary-tile/exploration-summary-tile.component.html:
```diff
@@ -1,135 +1,193 @@
+                  [isContainerNarrow]="isContainerNarrow"
+                  [isAddToPlaylistIconShown]="!isOwnedByCurrentUser">
+    </oppia-learner-dashboard-icons>
+    <mat-card class="mobile-activity-summary-card" *ngIf="!isRefresherExploration" [ngStyle]="{'margin': '5px'}">
```
Here and below, for ngStyle that are not dynamically changing CSS, can we define CSS classes and use those instead?

SubhashThenua replied:
Already written(https://github.com/SubhashThenua/oppia/blob/develop/core/templates/components/summary-tile/exploration-summary-tile.component.html#L92), but I have changed

kevintab95 commented on core/templates/components/summary-tile/exploration-summary-tile.component.html:
```diff
@@ -1,135 +1,193 @@
+    </div>
+    <div class="oppia-summary-section">
+      <div class="oppia-learner-dashboard-exploration-title">
+        {{explorationTitle}}
```
```suggestion
        {{ explorationTitle }}
```

kevintab95 replied:
Ditto elsewhere.

kevintab95 commented on core/templates/components/summary-tile/learner-story-summary-tile.component.html:
```diff
@@ -32,6 +32,61 @@
+    </div>
+    <div class="oppia-summary-section">
+      <div class="next-incomplete-chapter" *ngIf="isDisplayAreaHome()">
+        {{nextIncompleteNodeTitle}}
```
```suggestion
        {{ nextIncompleteNodeTitle }}
```

Ditto below.

kevintab95 commented on core/templates/components/summary-tile/learner-story-summary-tile.component.html:
```diff
@@ -163,4 +218,182 @@
     float: left;
   }
+
+/* redesigned learner story summary tile css*/
+
```
Is this necessary?

SubhashThenua replied:
Hi, @kevintab95 
Yes, This is necessary till this project is completed. I have to take care of both the old and redesigned learner dashboard and need to separate the code so that it is easy for me to remove the code of old learner dashboard. Also, I have to use the word **new** so that I have an idea of the code to whom it signifies(old one or new one). When I remove the code of the old learner dashboard, I will remove the **new** word also. 
Thanks!

kevintab95 replied:
The way to handle this would be to add a TODO and a corresponding issue describing what needs to be done. See examples in the codebase for TODOs.

kevintab95 commented on core/templates/components/summary-tile/learner-story-summary-tile.component.html:
```diff
@@ -163,4 +218,182 @@
+    text-align: left;
+  }
+  .new-learner-story-summary-tile .continue-button {
+    background: white 0% 0% no-repeat padding-box;
```
Ditto re colours.

SubhashThenua replied:
Done

kevintab95 commented on core/templates/components/summary-tile/learner-story-summary-tile.component.html:
```diff
@@ -163,4 +218,182 @@
+  .new-learner-story-summary-tile .oppia-topic-title {
+    color: #333;
+    display: block;
+    font-family: 'Roboto'sans-serif;
```
Is this correctly defined?

SubhashThenua replied:
Updated PTAL

kevintab95 commented on core/templates/pages/learner-dashboard-page/new-home-tab.component.html:
```diff
@@ -0,0 +1,148 @@
+<div class="new-home-tab">
```
Appending "new-*" here seems a bit odd. Will this be the learner dashboard page home tab going forward? If so, we should just call it home tab.

kevintab95 replied:
Please update the filename here and elsewhere too.

SubhashThenua replied:
Hi, @kevintab95 
I have created a file named **new-home-tab** and this file is having redesigned the home-tab. I have created a new file so that I can manage the old and new home tab. Once the project gets complete I will remove the old-home-tab file and keep the redesigned home tab, also I will remove the word **new**
Thanks!

kevintab95 commented on core/templates/components/summary-tile/learner-topic-summary-tile.component.html:
```diff
@@ -71,4 +95,51 @@
     height: 108px;
     width: 160px;
   }
+  /* redesigned learner topic summary tile css */
```
Ditto: Is this required?

SubhashThenua replied:
explained above PTAL.

kevintab95 replied:
Prefer TODO.

kevintab95 commented on core/templates/components/summary-tile/learner-story-summary-tile.component.html:
```diff
@@ -32,6 +32,61 @@
+      <div class="next-incomplete-chapter" *ngIf="isDisplayAreaHome()">
+        {{nextIncompleteNodeTitle}}
+      </div>
+      <div class="oppia-topic-title e2e-test-topic-name-in-new-new-learner-story-summary-tile">
```
typo: "..new-new.."

Also, here and elsewhere, please drop the "new-" part throughout the PR. Terminology is confusing, there is nothing "new" about it once merged. We're updating the look and feel of the dashboard and not maintaining 2 versions of it.

SubhashThenua replied:
Thanks for pointing this out, I have removed the typo and explained above why to use the new word.

kevintab95 commented on core/templates/pages/learner-dashboard-page/learner-groups-tab.component.html:
```diff
@@ -1,86 +1,62 @@
 <div class="learner-groups-tab" [ngStyle]="!windowIsNarrow && {'margin-top': '64px'}">
   <div class="learner-of-learner-groups-section">
-    <h3 class="learner-groups-section-heading">
+    <p class="learner-groups-section-heading">
```
Why use "p" for a heading? It seems semantically incorrect.

SubhashThenua replied:
Done as earlier.

kevintab95 commented on core/templates/pages/learner-dashboard-page/new-home-tab.component.html:
```diff
@@ -0,0 +1,148 @@
+
+
+  <div tabindex="0" class="topic-suggested-for-you untracked">
+    <p class="suggested-for-you-heading" [innerHTML]="'I18N_LEARNER_DASHBOARD_LEARN_SOMETHING_NEW_SECTION' | translate">
```
Ditto re HTML semantics

SubhashThenua replied:
Done

kevintab95 commented on scripts/typescript_checks.py:
```diff
@@ -151,6 +151,7 @@
     'core/templates/pages/exploration-player-page/services/question-player-engine.service.ts',
     'core/templates/pages/facilitator-dashboard-page/facilitator-dashboard-page.import.ts',
     'core/templates/pages/learner-dashboard-page/learner-dashboard-page.component.spec.ts',
+    'core/templates/pages/learner-dashboard-page/new-learner-dashboard.component.spec.ts',
```
Do not add files here. Please see the "NOTE TO DEVELOPERS" on top.

SubhashThenua replied:
Thanks for pointing out. I have created a new file for the learner dashboard named **new-leaner-dashboard**. I have created this file so that I can manage the old and new dashboard. I have just copy the code of **learner-dashboard.component.spec.ts** to ** new-learner-dashboard.component.spec.ts** and made changes. When I run the strict typescripts test, it is giving me an error logs due to the code of the old learner-dashboard file, so I have added this file to this list. Once this project is complete. I will shift the code from `new-learner-dashboard.component.spec.ts` to `learner-dashboard-page.component.spec.ts` and remove the `new-learner-dashboard-page.component.spec.ts`
Thanks!

U8NWXD replied:
I'm okay with this if @kevintab95 approves.

@SubhashThenua please be more careful and precise with your comments in the future. You refer to `learner-dashboard.component.spec.ts`, but I think you actually mean `learner-dashboard-page.component.spec.ts`. Similarly, you mention removing `new-learner-dashboard-page.component.spec.ts` when I think you mean `new-learner-dashboard.component.spec.ts`. These differences may seem trivial to you since you work with the frontend code a lot, but they're confusing for reviewers like me who don't work in the frontend much

kevintab95 replied:
Sounds good. Maybe add a TODO to remove this once the migration is complete?

kevintab95 commented on core/domain/exp_fetchers_test.py:
```diff
@@ -1013,3 +1013,23 @@ def test_migration_then_reversion_maintains_valid_exploration(self) -> None:
+        user_progress = exp_fetchers.get_user_progress_in_exploration(
+            user_id, [exp_id])
+
+        self.assertEqual(len(user_progress), 1)
+        self.assertEqual(user_progress[0]['exploration_id'], exp_id)
+        self.assertEqual(user_progress[0]['visited_checkpoints_count'], 1)
+        self.assertEqual(user_progress[0]['total_checkpoints_count'], 1)
```
Should there be pre-checks?

SubhashThenua replied:
I think line 1032 is working as a pre-check.

kevintab95 commented on core/domain/exp_fetchers.py:
```diff
@@ -673,3 +676,47 @@ def get_exploration_version_history(
+def get_user_progress_in_exploration(
+    user_id: str, exp_ids: List[str]
+) -> List[exp_domain.ExplorationProgressSummaryDict]:
+    """Returns the progress of multiple users in multiple exploration.
```
Isn't it just for a single user? I only see one user_id being passed?

SubhashThenua replied:
Yes, it's a single user. Updated PTAL

kevintab95 reviewed (changes requested):
Thanks @SubhashThenua! I took a pass on the HTML / CSS files and left some comments. PTAL

I will take a pass on the TS files soon but please feel free to address the existing comments meanwhile.

kevintab95 commented:
Hi @SubhashThenua, I just noticed that the lint checks are failing. Did they not run locally on pushing the latest commit?

Commit:
Merge remote-tracking branch 'upstream/develop' into Redesign-home-tab-GSoC-M1.3

SubhashThenua committed:
Merge branch 'develop' into Redesign-home-tab-GSoC-M1.3

SubhashThenua committed:
Update core/templates/components/summary-tile/collection-summary-tile.component.css

Co-authored-by: Kevin Thomas <kevintab@tutanota.com>

SubhashThenua committed:
Update core/templates/components/summary-tile/exploration-summary-tile.component.html

Co-authored-by: Kevin Thomas <kevintab@tutanota.com>

SubhashThenua committed:
Update core/templates/components/summary-tile/learner-story-summary-tile.component.html

Co-authored-by: Kevin Thomas <kevintab@tutanota.com>

Commit:
Merge remote-tracking branch 'upstream/develop' into Redesign-home-tab-GSoC-M1.3

SubhashThenua committed:
Merge branch 'develop' into Redesign-home-tab-GSoC-M1.3

Commit:
review changes by kevin

Commit:
Merge branch 'Redesign-home-tab-GSoC-M1.3' of https://github.com/SubhashThenua/oppia into Redesign-home-tab-GSoC-M1.3

Commit:
fix I18N

Commit:
fix test

SubhashThenua commented:
> Hi @SubhashThenua, I just noticed that the lint checks are failing. Did they not run locally on pushing the latest commit?

Hi @kevintab95 I have committed the changes suggested by Shivam, and due to this lint test get failed. Now I have fixed them.
Thanks!

Commit:
fix lint

Commit:
fix backend test

SubhashThenua committed:
Merge branch 'develop' into Redesign-home-tab-GSoC-M1.3

SubhashThenua committed:
Merge branch 'develop' into Redesign-home-tab-GSoC-M1.3

Commit:
added e2e

Commit:
Merge branch 'Redesign-home-tab-GSoC-M1.3' of https://github.com/SubhashThenua/oppia into Redesign-home-tab-GSoC-M1.3

Commit:
Merge remote-tracking branch 'upstream/develop' into Redesign-home-tab-GSoC-M1.3

SubhashThenua committed:
Merge branch 'develop' into Redesign-home-tab-GSoC-M1.3

Lawful2002 commented on core/templates/pages/learner-dashboard-page/new-home-tab.component.spec.ts:
```diff
@@ -0,0 +1,571 @@
+  };
+}
+
+describe('New-Home tab Component', () => {
```
```suggestion
describe('New Home tab Component', () => {
```

Lawful2002 commented on core/templates/pages/learner-dashboard-page/new-home-tab.component.spec.ts:
```diff
@@ -0,0 +1,571 @@
+    component.ngOnInit();
+    expect(component.completedExpIds).toEqual(completedExpIds);
+  });
+  it('should check total explortion Ids', () => {
```
```suggestion
  it('should check total explortion IDs', () => {
```

Lawful2002 commented on core/templates/pages/learner-dashboard-page/new-learner-dashboard.component.spec.ts:
```diff
@@ -0,0 +1,873 @@
+// Copyright 2021 The Oppia Authors. All Rights Reserved.
```
```suggestion
// Copyright 2021 The Oppia Authors. All Rights Reserved.
```
```suggestion
// Copyright 2023 The Oppia Authors. All Rights Reserved.
```

Lawful2002 reviewed (approved):
A few minor nits, otherwise LGTM!

SubhashThenua committed:
Update core/templates/pages/learner-dashboard-page/new-home-tab.component.spec.ts

Co-authored-by: Harshvardhan Singh <73544247+Lawful2002@users.noreply.github.com>

SubhashThenua committed:
Update core/templates/pages/learner-dashboard-page/new-home-tab.component.spec.ts

Co-authored-by: Harshvardhan Singh <73544247+Lawful2002@users.noreply.github.com>

SubhashThenua committed:
Update core/templates/pages/learner-dashboard-page/new-learner-dashboard.component.spec.ts

Co-authored-by: Harshvardhan Singh <73544247+Lawful2002@users.noreply.github.com>

SubhashThenua committed:
Merge branch 'develop' into Redesign-home-tab-GSoC-M1.3

aasiffaizal commented on core/domain/exp_fetchers.py:
```diff
@@ -673,3 +676,47 @@ def get_exploration_version_history(
+        summaries of the user corresponding to all exploration.
+    """
+    exp_id_to_exp_map = exp_fetchers.get_multiple_explorations_by_id(exp_ids)
+    user_id_exp_id_combinations = list(itertools.product([user_id], exp_ids))
```
```suggestion
    user_id_exp_id_combinations = itertools.product([user_id], exp_ids)
```

aasiffaizal replied:
Is iterating to a list necessary?

SubhashThenua replied:
Thanks, @aasiffaizal for the review, yes we need a list that contains user_id and exp_id cartesian product so that I can fetch the data from the database using the get multi function.

aasiffaizal replied:
Yeah but do we need to use the list function?
I mean wouldn't it work without that since you are iterating through it down in the for loop

U8NWXD reviewed (approved):
Approving since @kevintab95 already left a blocking review on my only concern, and I'm okay with leaving it as-is if they approve

U8NWXD commented:
Replacing @srijanreddy98 with @seanlip for reviews since @srijanreddy98 is away

seanlip commented on core/controllers/learner_dashboard.py:
```diff
@@ -303,3 +304,52 @@ def get(self) -> None:
+        }
+    }
+
+    # TODO(#16539): Change the type of value that is rendered to JSON,
```
Don't do this. Just render a dict with a single key-value pair. The key can be 'exploration_progress' and the value is the list in question.

Don't introduce new deviations which require a TODO when this is not needed.

SubhashThenua replied:
Hi, @seanlip Thanks for the review,

- I need to take all three variables (exp_id, visited checkpoints, total checkpoints). Because I am using them in the front end. Also, It is already used in many places in the code base please see here (https://github.com/SubhashThenua/oppia/blob/develop/core/controllers/learner_group.py#L1072).
Thanks!

seanlip replied:
Yes, read my comment more carefully please. It still allows you to send all the information you want. I'm talking about adding an outer dict to the response, i.e.

```
{
    'exploration_progress': exploration_progress
}
```

so that you don't need to introduce a TODO.

Also, just because a TODO exists elsewhere is not a reason to add more TODOs in new code that you write. That just ends up leaving more stuff for people to clean up. So please don't do it in future.

seanlip commented on core/domain/exp_fetchers.py:
```diff
@@ -673,3 +676,47 @@ def get_exploration_version_history(
+def get_user_progress_in_exploration(
+    user_id: str, exp_ids: List[str]
+) -> List[exp_domain.ExplorationProgressSummaryDict]:
+    """Returns the progress of user in multiple exploration.
```
user --> the user
exploration --> explorations

SubhashThenua replied:
Done PTAL

seanlip commented on core/domain/exp_fetchers.py:
```diff
@@ -673,3 +676,47 @@ def get_exploration_version_history(
+
+    Args:
+        user_id: str. The user id of the user.
+        exp_ids: list(str). The ids of the exploration.
```
explorations

SubhashThenua replied:
Done PTAL

seanlip commented on core/domain/exp_fetchers.py:
```diff
@@ -673,3 +676,47 @@ def get_exploration_version_history(
+
+    Returns:
+        list(ExplorationProgressSummaryDict). The list of the progress
+        summaries of the user corresponding to all exploration.
```
explorations

SubhashThenua replied:
Done PTAL

seanlip reviewed (changes requested):
Just a few comments. Please reassign me when they're addressed. Thanks!

heyimShivam commented on core/templates/components/summary-tile/collection-summary-tile.component.ts:
```diff
@@ -50,13 +53,17 @@ export class CollectionSummaryTileComponent implements OnInit, OnDestroy {
   @Input() isOwnedByCurrentUser: boolean = false;
   @Input() mobileCutoffPx!: number;
+  @Input() displayAreaIsLearnerDashboard: boolean = false;
+
 
```
not required

SubhashThenua replied:
Done

seanlip commented:
Also, I took a look at the UI. At 0:07, it seems there's an issue with the navbar?

![Screenshot from 2023-07-31 23-12-15](https://github.com/oppia/oppia/assets/10575562/578f611a-238c-4b2a-90f6-eb3185ead35b)

heyimShivam commented on core/templates/components/summary-tile/exploration-summary-tile.component.css:
```diff
@@ -69,3 +70,168 @@ oppia-exploration-summary-tile .views-text {
 oppia-exploration-summary-tile .thumbnail-image {
   display: block;
 }
+
```
not required

SubhashThenua replied:
Done

heyimShivam commented on core/templates/components/summary-tile/learner-story-summary-tile.component.ts:
```diff
@@ -47,8 +50,14 @@ export class LearnerStorySummaryTileComponent implements OnInit {
   openInNewWindow = false;
+  circumference = (18 * 2 * Math.PI);
+  getStrokeDashOffSet!: number;
+
```
new line not required.

SubhashThenua replied:
Done

heyimShivam commented on core/templates/components/summary-tile/learner-story-summary-tile.component.ts:
```diff
@@ -131,11 +146,25 @@ export class LearnerStorySummaryTileComponent implements OnInit {
+    return '';
+  }
+
+
```
new line not required

SubhashThenua replied:
Done

heyimShivam commented on core/templates/pages/learner-dashboard-page/carousel-bar.component.ts:
```diff
@@ -0,0 +1,137 @@
+ }
+
+ ngAfterViewInit(): void {
+   setTimeout(() => {
+     this.isScrollable = this.initCarousel();
+   });
```
Please add a comment on why setTimeout is used here.

SubhashThenua replied:
Done

heyimShivam commented on core/templates/pages/learner-dashboard-page/learner-dashboard-page.component.spec.ts:
```diff
@@ -119,6 +127,8 @@ describe('Learner dashboard page', () => {
   let learnerGroupBackendApiService: LearnerGroupBackendApiService;
   let urlService: UrlService;
+  let mockPlatformFeatureService = new MockPlatformFeatureService();
+
```
new line not required

SubhashThenua replied:
Done

heyimShivam commented on core/templates/pages/learner-dashboard-page/new-home-tab.component.ts:
```diff
@@ -0,0 +1,263 @@
+    this.goalTopicsLength = this.goalTopics.length;
+
+    if (allGoals.length !== 0) {
+      var allGoalIds = [];
+      for (var goal of allGoals) {
+        allGoalIds.push(goal.id);
+      }
+      var uniqueGoalIds = Array.from(new Set(allGoalIds));
+      for (var uniqueGoalId of uniqueGoalIds) {
+        var index = allGoalIds.indexOf(uniqueGoalId);
+        this.continueWhereYouLeftOffList.push(allGoals[index]);
+      }
+    }
```
why let is not be used in the place of var? (as per as the previous review)

SubhashThenua replied:
Done

heyimShivam commented on core/templates/pages/learner-dashboard-page/new-home-tab.component.ts:
```diff
@@ -0,0 +1,263 @@
+  ) {}
+
+  ngOnInit(): void {
+    var allGoals = [...this.currentGoals, ...this.partiallyLearntTopicsList];
```
why let is not used?

SubhashThenua replied:
Done

heyimShivam commented:
![Screenshot from 2023-07-31 21-01-15](https://github.com/oppia/oppia/assets/77006897/2f5a9948-d196-4b79-ae81-a5df7a683e42)

issue with the navbar

SubhashThenua committed:
Merge branch 'develop' into Redesign-home-tab-GSoC-M1.3

Commit:
added e2e for home tab of new-leanrer-dashboard

Commit:
Merge branch 'Redesign-home-tab-GSoC-M1.3' of https://github.com/SubhashThenua/oppia into Redesign-home-tab-GSoC-M1.3

Commit:
Merge remote-tracking branch 'upstream/develop' into Redesign-home-tab-GSoC-M1.3

aasiffaizal commented on core/domain/exp_fetchers.py:
```diff
@@ -673,3 +676,47 @@ def get_exploration_version_history(
+    all_exploration_progress: List[
+        exp_domain.ExplorationProgressSummaryDict] = []
+    for i, user_id_exp_id_pair in enumerate(user_id_exp_id_combinations):
+        exp_id = user_id_exp_id_pair[1]
```
```suggestion
        _, exp_id = user_id_exp_id_pair
```

SubhashThenua replied:
Done

aasiffaizal commented on core/domain/exp_fetchers.py:
```diff
@@ -673,3 +676,47 @@ def get_exploration_version_history(
+        model = exp_user_data_models[i]
+        visited_checkpoints = 0
+        if model is not None:
+            most_recently_visited_checkpoint = (
+                model.most_recently_reached_checkpoint_state_name)
```
```suggestion
            most_recently_visited_checkpoint = (
                model.most_recently_reached_checkpoint_state_name
            )
```

SubhashThenua replied:
> ![Screenshot from 2023-07-31 21-01-15](https://user-images.githubusercontent.com/77006897/257270856-2f5a9948-d196-4b79-ae81-a5df7a683e42.png)
> 
> issue with the navbar

Replied to sean comment  PTAL

aasiffaizal commented on core/domain/exp_fetchers.py:
```diff
@@ -673,3 +676,47 @@ def get_exploration_version_history(
+            most_recently_visited_checkpoint = (
+                model.most_recently_reached_checkpoint_state_name)
+            if most_recently_visited_checkpoint is not None:
+                visited_checkpoints = all_checkpoints.index(
+                    most_recently_visited_checkpoint) + 1
+        all_exploration_progress.append({
```
```suggestion
                visited_checkpoints = all_checkpoints.index(
                    most_recently_visited_checkpoint
                ) + 1
        all_exploration_progress.append({
```

aasiffaizal reviewed (changes requested):
Left a few comments @SubhashThenua

SubhashThenua committed:
Merge branch 'develop' into Redesign-home-tab-GSoC-M1.3

Commit:
review changes

SubhashThenua commented:
> Also, I took a look at the UI. At 0:07, it seems there's an issue with the navbar?
> 
> ![Screenshot from 2023-07-31 23-12-15](https://user-images.githubusercontent.com/10575562/257266242-578f611a-238c-4b2a-90f6-eb3185ead35b.png)

Thanks @seanlip for review, This issue is not due to my changes, it is already on develop
Please see:
![image](https://github.com/oppia/oppia/assets/100484672/567097cf-669b-49cc-8c7c-fbd3cd02d6e4)
Thanks!

Commit:
Merge branch 'Redesign-home-tab-GSoC-M1.3' of https://github.com/SubhashThenua/oppia into Redesign-home-tab-GSoC-M1.3

SubhashThenua committed:
Merge branch 'develop' into Redesign-home-tab-GSoC-M1.3

SubhashThenua committed:
Merge branch 'develop' into Redesign-home-tab-GSoC-M1.3

kevintab95 commented on core/templates/pages/learner-dashboard-page/learner-dashboard-page.component.ts:
```diff
@@ -137,6 +138,7 @@ export class LearnerDashboardPageComponent implements OnInit, OnDestroy {
   progressImageUrl: string = '';
   windowIsNarrow: boolean = false;
   directiveSubscriptions = new Subscription();
+  SHOW_REDESIGNED_LEARNER_DASHBOARD: boolean = false;
```
Where is this used?

SubhashThenua replied:
Thanks @kevintab95  for the review. I have removed it as unused. Sometimes I think of the approach and apply and when it is not working I remove the code. I think this remains from that only.

kevintab95 commented on core/templates/pages/learner-dashboard-page/learner-dashboard-page.component.ts:
```diff
@@ -262,6 +265,12 @@ export class LearnerDashboardPageComponent implements OnInit, OnDestroy {
     this.directiveSubscriptions.unsubscribe();
   }
 
+  isShowRedesignedLearnerDashboardFeatureFlagEnable(): boolean {
```
```suggestion
  isRedesignedLearnerDashboardEnabled(): boolean {
```

SubhashThenua replied:
Done

kevintab95 commented on core/templates/app.constants.ts:
```diff
@@ -238,4 +238,5 @@ export const AppConstants = {
   CONTRIBUTOR_BADGE_INITIAL_LEVELS: [1, 10, 20, 50, 100, 200, 300, 400, 500],
   CONTRIBUTOR_CERTIFICATE_LOGO:
     '/assets/images/contributor_dashboard/oppia-logo.jpg',
+  LEARN_SOMETHING_NEW_TILE_WIDTH_PX: 216
```
Where is this used?

SubhashThenua replied:
Removed as unused

kevintab95 commented on core/templates/components/summary-tile/collection-summary-tile.component.ts:
```diff
@@ -127,6 +134,13 @@ export class CollectionSummaryTileComponent implements OnInit, OnDestroy {
     );
   }
 
+  isButtonHovered(): string {
+    if (this.buttonIsHovered) {
+      return 'box-shadow: 0 5px 5px -5px #333;';
+    }
+    return '';
+  }
```
I see this code being duplicated in several places. Can we create a class for the box-shadow rule and apply that using ngClass in all the places this is used?

SubhashThenua replied:
Done PTAL

kevintab95 commented on core/templates/components/summary-tile/exploration-summary-tile.component.spec.ts:
```diff
@@ -251,6 +251,7 @@ describe('Exploration Summary Tile Component', () => {
     expect(userServiceSpy).toHaveBeenCalled();
     expect(windowResizeSpy).toHaveBeenCalled();
     expect(windowWidthSpy).toHaveBeenCalled();
+    flush();
```
Why is flush required now?

SubhashThenua replied:
Done PTAL

kevintab95 replied:
Not done.

I'm stopping review here because it looks like my previous comments have not been addressed.

kevintab95 commented on core/templates/components/summary-tile/exploration-summary-tile.component.ts:
```diff
@@ -91,6 +94,10 @@ export class ExplorationSummaryTileComponent implements OnInit, OnDestroy {
   expTitleTranslationKey!: string;
   expObjectiveTranslationKey!: string;
+  buttonIsHovered: boolean = false;
+  circumference = (18 * 2 * Math.PI);
```
Please avoid duplicating code. Can we create a constant and reuse it?

SubhashThenua replied:
Done PTAL

kevintab95 commented on core/templates/components/summary-tile/exploration-summary-tile.component.ts:
```diff
@@ -91,6 +94,10 @@ export class ExplorationSummaryTileComponent implements OnInit, OnDestroy {
   expObjectiveTranslationKey!: string;
+  buttonIsHovered: boolean = false;
+  circumference = (18 * 2 * Math.PI);
+  getStrokeDashOffSet!: number;
```
Is this a method or a function? Please see [code style guidelines](https://github.com/oppia/oppia/wiki/Coding-style-guide#javascript) regarding naming.

SubhashThenua replied:
Done PTAL

kevintab95 commented on core/templates/components/summary-tile/exploration-summary-tile.component.ts:
```diff
@@ -250,11 +262,24 @@ export class ExplorationSummaryTileComponent implements OnInit, OnDestroy {
   }
 
+  getStrokeDashOffSetValue(explorationProgress: number): number {
+    let segmentLength =
+    (this.circumference - (explorationProgress / 100) * this.circumference);
+    return segmentLength;
```
Can we do this inline in the HTML?

SubhashThenua replied:
I am using the function in ngOnChanges that contains this code. I think we need to keep this as it is.

kevintab95 replied:
Sorry, I don't quite understand.. it seems like the only value that changes would be explorationProgress (circumference is a constant), is that not getting updated in the HTML without ngOnChanges?

kevintab95 commented on core/templates/domain/exploration/read-only-exploration-backend-api.service.spec.ts:
```diff
@@ -452,4 +454,48 @@ describe('Read only exploration backend API service', () => {
     expect(roebas.isCached('0')).toBe(false);
   }));
+
+  it('should successfully fetch learners progress in stories chapters',
```
```suggestion
  it('should successfully fetch learners progress in chapters',
```

SubhashThenua replied:
Done PTAL

kevintab95 commented on core/templates/domain/exploration/read-only-exploration-backend-api.service.ts:
```diff
@@ -237,6 +241,34 @@ export class ReadOnlyExplorationBackendApiService {
+   * Retrives the Progress of exploratin
+   * chapters are nothing but exploration.
+   * The exploration used in story are called chapters.
+   * To reduce the code redundacy, ChapterProgressSummay Model is used.
```
A bit confusing to read.. can we just say "Retrieves progress in a chapters."

and change the function name to `fetchChapterProgressSummaries`?

SubhashThenua replied:
Done PTAL

kevintab95 commented on core/templates/domain/state_card/state-card.model.ts:
```diff
@@ -302,7 +302,7 @@ export class StateCard {
     let writtenTranslation = entityTranslation.getWrittenTranslation(
       this._contentId) as TranslatedContent;
-    if (writtenTranslation) {
+    if (writtenTranslation && !writtenTranslation.needsUpdate) {
```
Why did this change?

SubhashThenua replied:
It is added when the beam job PR of Kshitij gets merged into develop before testing on develop and I have manually reverted the changes and it left from it. I have removed it as unused.
Thanks!

kevintab95 commented on core/templates/pages/learner-dashboard-page/carousel-bar.component.css:
```diff
@@ -0,0 +1,67 @@
+.oppia-carousel-bar .oppia-carousel-scroller {
+  align-items: center;
+  border-radius: 4px;
+  color: white;
```
Use HEX.

SubhashThenua replied:
Done PTAL

kevintab95 commented on core/templates/pages/learner-dashboard-page/carousel-bar.component.spec.ts:
```diff
@@ -0,0 +1,331 @@
+    expect(component.isScrollable).toBe(true);
+  });
+
+  it('should scroll untraked topics to right, when' +
```
untraked -> untracked

kevintab95 replied:
Ditto elsewhere.

SubhashThenua replied:
Done PTAL

kevintab95 commented on core/templates/pages/learner-dashboard-page/new-learner-dashboard.component.ts:
```diff
@@ -0,0 +1,385 @@
+
+  decodePngURIData(base64ImageData: string): string {
+    return decodeURIComponent(base64ImageData);
+  }
```
Where is this used?

SubhashThenua replied:
Removed as unused

SubhashThenua replied:
Removed as unused

kevintab95 commented on core/templates/pages/learner-dashboard-page/new-learner-dashboard.component.ts:
```diff
@@ -0,0 +1,385 @@
+
+  getLocaleAbbreviatedDatetimeString(millisSinceEpoch: number): string {
+    return this.dateTimeFormatService.getLocaleAbbreviatedDatetimeString(
+      millisSinceEpoch);
```
Where is this used?

Please remove unused code from this file.

SubhashThenua replied:
Removed as unused

kevintab95 commented on core/templates/pages/learner-dashboard-page/new-learner-dashboard.component.spec.ts:
```diff
@@ -0,0 +1,873 @@
+    });
+
+    it('should get human readable status from provided status', () => {
+      expect(component.getHumanReadableStatus('open')).toBe('Open');
+      expect(component.getHumanReadableStatus('compliment')).toBe('Compliment');
+      expect(component.getHumanReadableStatus('not_actionable')).toBe(
```
Looks like these are unused code. Can we remove the unused code instead?

SubhashThenua replied:
Removed as unused

kevintab95 commented on core/templates/pages/learner-dashboard-page/new-learner-dashboard.component.css:
```diff
@@ -0,0 +1,371 @@
+
+.new-learner-dashboard .oppia-learner-dashboard-main-menu {
+  background-color: #00645c;
+  color: white;
```
Use HEX.

SubhashThenua replied:
Done

kevintab95 commented on core/templates/pages/learner-dashboard-page/new-learner-dashboard.component.css:
```diff
@@ -0,0 +1,371 @@
+
+.new-learner-dashboard .oppia-learner-dashboard-menu {
+  align-items: center;
+  border: 2px solid white;
```
Use HEX.

SubhashThenua replied:
Done

kevintab95 commented on core/templates/pages/learner-dashboard-page/new-learner-dashboard.component.css:
```diff
@@ -0,0 +1,371 @@
+/*
+  NOTE to developers: We generate RTL CSS from this file in the webpack
+  compilation, here are sme additional rules that can be added to the CSS files:
```
Please fix the typo here: sme -> some.

No need to change it in other places as part of this PR.

SubhashThenua replied:
Done

kevintab95 commented on core/templates/pages/learner-dashboard-page/learner-groups-tab.component.css:
```diff
@@ -236,12 +259,30 @@
+}
+@media (max-width: 767px) {
+  .learner-groups-tab .learner-groups-section-heading {
+    font-family: Capriola,Roboto,Arial,sans-serif;
```
Not done. As part of my previous review, I had required the spacing here be fixed.

SubhashThenua replied:
Done, sorry I missed it

kevintab95 commented on core/templates/pages/learner-dashboard-page/learner-groups-tab.component.css:
```diff
@@ -84,19 +86,36 @@
+  align-items: center;
+  background: #00645c;
+  border-radius: 4px;
+  color: white;
```
Not done. Please use HEX.

SubhashThenua replied:
Done

kevintab95 commented on core/templates/pages/learner-dashboard-page/learner-groups-tab.component.css:
```diff
@@ -32,9 +34,11 @@
 .learner-groups-tab .learner-groups-section-heading {
-  color: #00645c;
+  color: #205d86;
+  font-family: Capriola,Roboto,Arial,sans-serif;
```
Fix spaces.

SubhashThenua replied:
Done

kevintab95 commented on core/templates/pages/learner-dashboard-page/carousel-bar.component.spec.ts:
```diff
@@ -0,0 +1,331 @@
+    expect(component.disableRightButton).toBe(false);
+  });
+
+  it('should disable left button' +
+   ' when carouselScrollPositionPx is less than or equal to 0', () => {
```
What does this test ensure for the user? The test description should not discuss implementation specifics. Please see https://github.com/oppia/oppia/wiki/Frontend-tests#name-tests-clearly

kevintab95 replied:
Ditto below.

kevintab95 commented on core/templates/pages/learner-dashboard-page/new-home-tab.component.css:
```diff
@@ -0,0 +1,400 @@
+}
+
+.new-home-tab .community-lessons-playlist-section .remove-icon {
+  color: white;
```
Use Hex.

SubhashThenua replied:
Done

kevintab95 commented on core/templates/pages/learner-dashboard-page/new-home-tab.component.spec.ts:
```diff
@@ -0,0 +1,571 @@
+  });
+
+  it('should check the completed ExpIds', () => {
+    const completed = {
```
```suggestion
    const completedLearnerExplorationSummaryDict = {
```

SubhashThenua replied:
Done

kevintab95 commented on core/templates/pages/learner-dashboard-page/new-home-tab.component.spec.ts:
```diff
@@ -0,0 +1,571 @@
+    expect(component.storyInProgress.length).toBe(1);
+  });
+
+  it('should check the completed ExpIds', () => {
```
Not sure what this test does based on the description. Please rewrite it per https://github.com/oppia/oppia/wiki/Frontend-tests#name-tests-clearly

SubhashThenua replied:
Done PTAL

kevintab95 commented on core/templates/pages/learner-dashboard-page/new-home-tab.component.spec.ts:
```diff
@@ -0,0 +1,571 @@
+    component.ngOnInit();
+    expect(component.completedExpIds).toEqual(completedExpIds);
+  });
+  it('should check total explortion IDs', () => {
```
Typo.

Perhaps rename to "should populate exploration ID count in learner exploration summary"

SubhashThenua replied:
Done PTAL

kevintab95 commented on core/templates/pages/learner-dashboard-page/new-home-tab.component.spec.ts:
```diff
@@ -0,0 +1,571 @@
+    expect(component.completedExpIds).toEqual(completedExpIds);
+  });
+  it('should check total explortion IDs', () => {
+    const exp1 = {
```
Unclear what "exp1" means. Please use a more descriptive name so it is clear what data it represents.

SubhashThenua replied:
Done PTAL

kevintab95 commented on core/templates/pages/learner-dashboard-page/new-home-tab.component.spec.ts:
```diff
@@ -0,0 +1,571 @@
+            result: Promise.resolve('success')
+          }) as NgbModalRef;
+      });
+      const exp1 = {
```
Ditto: Please use a more meaningful name.

SubhashThenua replied:
Done PTAL

kevintab95 commented on core/templates/pages/learner-dashboard-page/new-home-tab.component.ts:
```diff
@@ -0,0 +1,263 @@
+    return Object.keys(object).length !== 0;
+  }
+
+  getClassroomLink(classroomUrlFragment: string): string {
```
Prefer URL over Link to maintain consistency in the codebase.

SubhashThenua replied:
I have to use this as a function. Because I am creating a link using Url Interpolation. Can you Please check the function.
Already used (https://github.com/oppia/oppia/blob/develop/core/templates/pages/learner-dashboard-page/home-tab.component.ts#L100). PTAL

kevintab95 replied:
I meant using `getClassroomURL` instead of `getClassroomLink`.

kevintab95 reviewed (changes requested):
I've done a full review of the PR. Few notes:
1. Please only reply "done" when you have addressed the comment by making a code change.
2. Reuse code where possible.
3. Avoid copy-pasting code. If you do have to, please only copy what is required and leave out any unused code.

Thanks!

seanlip commented on core/tests/webdriverio_desktop/newLearnerDashboard.js:
```diff
@@ -0,0 +1,285 @@
+    await users.logout();
+  });
+
+  it('should create a leaner group', async function() {
```
learner

Commit:
fix e2e

Commit:
change flag to prod

SubhashThenua committed:
Update platform_feature_list.py

Commit:
review changes

SubhashThenua committed:
Merge branch 'develop' into Redesign-home-tab-GSoC-M1.3

SubhashThenua committed:
Update carousel-bar.component.spec.ts

aasiffaizal commented on core/domain/exp_fetchers.py:
```diff
@@ -673,3 +676,49 @@ def get_exploration_version_history(
+                model.most_recently_reached_checkpoint_state_name
+                )
+            if most_recently_visited_checkpoint is not None:
+                visited_checkpoints = all_checkpoints.index(
+                    most_recently_visited_checkpoint
+                    ) + 1
```
```suggestion
                visited_checkpoints = all_checkpoints.index(
                    most_recently_visited_checkpoint
                ) + 1
```

aasiffaizal commented on core/domain/exp_fetchers.py:
```diff
@@ -673,3 +676,49 @@ def get_exploration_version_history(
+        model = exp_user_data_models[i]
+        visited_checkpoints = 0
+        if model is not None:
+            most_recently_visited_checkpoint = (
+                model.most_recently_reached_checkpoint_state_name
+                )
```
```suggestion
            most_recently_visited_checkpoint = (
                model.most_recently_reached_checkpoint_state_name
            )
```

heyimShivam commented:
Hi @SubhashThenua 
1. please fix the search bar padding and margin for different screen sizes (in all places).

https://github.com/oppia/oppia/assets/77006897/a12e834b-e437-43c5-b199-e63e17ef893e



2. Please fix the navbar buttons' padding
![1](https://github.com/oppia/oppia/assets/77006897/18242587-ff53-4164-b3d2-28616e0d0d7f)

3.  In the Exploration editor page the navbar has been changed in e2e testing video.
![Screenshot from 2023-08-04 04-05-48](https://github.com/oppia/oppia/assets/77006897/9a94b7a6-bb32-48c4-ba4e-ff691c0f15b1)

4. there are some extra shadows behind the text in the groups section.
![Screenshot from 2023-08-04 03-25-40](https://github.com/oppia/oppia/assets/77006897/f4fff43f-8548-4033-b877-fd4844879abf)

5. In the E2E test checkpoint features, percentage features, carousel, and resume where you left, still seem not tested please make sure they are tested as well if not.

kevintab95 commented on core/templates/components/summary-tile/exploration-summary-tile.component.ts:
```diff
@@ -103,6 +110,11 @@ export class ExplorationSummaryTileComponent implements OnInit, OnDestroy {
   ) {}
 
+  ngOnChanges(changes: SimpleChanges): void {
+    this.strokeDashOffSet =
```
Wrap in `(..)` if a statement need to be broken into multiple lines.

kevintab95 commented on core/templates/domain/exploration/read-only-exploration-backend-api.service.ts:
```diff
@@ -237,6 +241,31 @@ export class ReadOnlyExplorationBackendApiService {
   }
 
+  /**
+   * Retrives the Progress in a chapters
```
```suggestion
   * Retrives the progress in chapters.
```

Apologies, I made a typo earlier. Drop the "a".

kevintab95 commented on scripts/typescript_checks.py:
```diff
@@ -26,6 +26,10 @@
 from typing import List, Optional, Sequence
 
 # TODO(#15567): This can be removed after Literal in utils.py is loaded
+# TODO(#18731): File new-learner-dashboard.component.spec.ts' will be
+# removed from the TS_STRICT_EXCLUDE_PATHS once the learner Dashboard
+# completely redesign and the code of this file will be shifted to
+# learner-dashboard-page.component.spec.ts.
```
This TODO is in the middle of another TODO.. 

Should we put it right above the specific line that was added?

kevintab95 commented on core/templates/pages/learner-dashboard-page/carousel-bar.component.spec.ts:
```diff
@@ -0,0 +1,331 @@
+    expect(component.disableRightButton).toBe(false);
+  });
+
+  it('should disable left button' +
+   ' when carouselScrollPosition is at left most position', () => {
```
Not done. See https://github.com/oppia/oppia/pull/18594#discussion_r1281257238

kevintab95 commented on core/templates/pages/learner-dashboard-page/new-home-tab.component.spec.ts:
```diff
@@ -0,0 +1,572 @@
+    expect(component.storyInProgress.length).toBe(1);
+  });
+
+  it('should check the number of completed exploration list', () => {
```
```suggestion
  it('should correctly populate the list of completed explorations', () => {
```

kevintab95 commented on core/templates/pages/learner-dashboard-page/new-home-tab.component.spec.ts:
```diff
@@ -0,0 +1,572 @@
+      component.explorationPlaylist = [expSummary];
+      component.ngOnInit();
+      expect(component.totalExploration).toEqual([expSummary]);
+    });
```
nit: newline

kevintab95 commented on core/templates/pages/learner-dashboard-page/new-home-tab.component.spec.ts:
```diff
@@ -0,0 +1,572 @@
+  });
+  it('should populate exploration ID count in learner exploration summary',
+    () => {
+      const exploration1 = {
```
What does the "1" here signify? Avoid appending numbers to variables unless they convey some meaning. There is only one exploration in this test, so just "exploration" might do.

kevintab95 replied:
Ditto below in this file.
